File: backend/app/models/task.py
from pydantic import BaseModel
from typing import Optional, Literal
from datetime import date, datetime


class TaskCreate(BaseModel):
    title: str
    description: Optional[str] = None
    due_date: Optional[date] = None
    priority: Optional[Literal["low", "medium", "high"]] = None
    course_id: Optional[str] = None


class TaskUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    due_date: Optional[date] = None
    priority: Optional[Literal["low", "medium", "high"]] = None
    completed: Optional[bool] = None
    course_id: Optional[str] = None


class Task(BaseModel):
    id: str
    user_id: Optional[str] = None
    course_id: Optional[str] = None
    title: str
    description: Optional[str] = None
    due_date: Optional[date] = None
    completed: bool = False
    priority: Optional[Literal["low", "medium", "high"]] = None
    created_at: Optional[datetime] = None

File: frontend/src/types/task.ts
export type TaskPriority = 'low' | 'medium' | 'high';

export interface Task {
  id: string;
  user_id: string | null;
  course_id: string | null;
  title: string;
  description: string | null;
  due_date: string | null;
  completed: boolean;
  priority: TaskPriority | null;
  created_at: string | null;
}

export interface TaskPayload {
  title: string;
  description?: string | null;
  due_date?: string | null;
  priority?: TaskPriority | null;
  course_id?: string | null;
}
